### ISSUE TITLE
Color fringe artifacts on whole image when inpainting->save->load->inpaint->etc multiple times in a row

### ISSUE THREAD

**Issue by RandomGitUser321** [comfyanonymous/ComfyUI#1841]
Even though I'm using masking, the whole image will progressively get more blown out with color fringing/chromatic aberration looking artifacts.

Here's the basic gist of my workflow:
Make something
Save it to png
Load it into an image node
Mask something off
Sample to change masked area until I get what I want
Save it again
Load that new save into an image node
Mask off new regions
etc etc

In terms of what I'm using, I'm just jusing the base SDXL 1.0 base, sometimes the SDXL 1.0 refiner(usually not) and the SDXL 1.0 vae. I've tested the issue with regular masking->vae encode->set latent noise mask->sample and I've also tested it with the load unet SDXL inpainting 0.1 model->mask->vae encode for inpainting-sample. In terms of samplers, I'm just using dpm++ 2m karras and usually around 25-32 samples, but that shouldn't be causing the rest of the unmasked image to be touched anyways.

After a few or so round-trips of that, the rest of the untouched image will progressively get worse and worse. I'm thinking that maybe it's a floating point rounding issue that keeps rounding upward? Or maybe it has something to do with the vae encode->vae decode portion like a precision/rounding issue in there? I can partially mitigate it by using something like Ultimate SD Upscale with a low denoise value like 0.25 or 0.3, but I'd rather not have to do that every single loop through.

Any ideas?

**Comment by jn-jairo** [comment]
For masked image to image you need to use the ImageCompositeMasked node after to filter the masked area, I don't know what, but something alter a little bit the parts not masked, for one generation it isn't that noticeable but if you do multiple generations you will see it.

I noticed it on my workflow for upscaled inpaint of masked areas, without the ImageCompositeMasked there is a clear seam on the upscaled square, showing that the whole square image was altered, not just the masked area, but adding the ImageCompositeMasked solved the problem, making a seamless inpaint.

**Comment by jn-jairo** [comment]
> > ImageCompositeMasked node
> 
> Thanks, I'll definitely try! Sounds like it's just overlaying the original image over top of the masked resample output minus the masked portion, which would definitely work as a crutch. But that still sounds like a bandaid fix for something else going on under the hood with precision or rounding.

My guess is the VAE, if you just Load Image > VAE Encode > VAE Decode > Save Image the saved image is different from the loaded one.

### Input
![the-mona-lisa](https://github.com/comfyanonymous/ComfyUI/assets/5104869/44e71dbf-68f8-4234-ad4c-ae13ee139dcc)

### Output
![ComfyUI_temp_lpczn_00001_](https://github.com/comfyanonymous/ComfyUI/assets/5104869/4cf7c3be-de60-48a2-827b-abd2835f39e6)

### Difference
![the-mona-lisa-diff](https://github.com/comfyanonymous/ComfyUI/assets/5104869/eaef0014-2c76-493d-862e-02848f7ff433)

**Comment by jn-jairo** [comment]
> Yep, sounds like some rounding/precision issue in there then. At any rate, I'll just work around it for now. I'm going to try testing it in A1111 to see if this issue is present there as well. Maybe it's just some kind of limitation of the whole vae encode->latent space->decode system. I don't know enough about the whole process though to really guess anything beyond that though.

I looked at the A1111 code and it does a process similar to the ImageCompositeMasked to get the output image when inpainting, so I guess that is how SD works and the ImageCompositeMasked is mandatory for a good result.

**Comment by jn-jairo** [comment]
I blur the mask by converting it to image and using the ImageBlur node and convert back to mask, no hard seams with this process, like this:

![image](https://github.com/comfyanonymous/ComfyUI/assets/5104869/1dff734d-9121-4405-ac02-1395ed731fda)

I also made an ImageCrop node to crop by mask (I copied it from A1111), and a ImageUncrop (it's just the ImageCompositeMasked with different parameters for the area), so it's faster to inpaint a small area and recombine it.

![image](https://github.com/comfyanonymous/ComfyUI/assets/5104869/425b03fc-e1dc-40cd-b930-f6345a93791d)

**Comment by jn-jairo** [comment]
> Guess that yeah the artifacts come from being compressed down into the latent 256x256 space and then scaled back to 1024x1024

Yeah, you have to lose something when compressing that much, the latent is even smaller because it divides by 8 so a 512x512 pixels image converts to a 64x64 in the latent space, it is a huge compression, and that makes the SD fast.

**Comment by ancillarymagnet** [comment]
You can fix this issue by using:
VAE Encode (not the inpaint one) -> Set Latent Noise Mask

![image](https://github.com/comfyanonymous/ComfyUI/assets/9907697/8f6388ba-4040-4157-adb2-7e2e75c1d3eb)
